### PR TITLE
STYLE: fix pylint unnecessary-comprehension warnings

### PR DESCRIPTION
**Pull request by Moisan**
Related to https://github.com/pandas-dev/pandas/issues/48855
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Review by rhshadrach — APPROVED** [review]
lgtm, ping on green

File: pandas/tests/groupby/test_grouping.py [hunk 1]
```diff
@@ -946,7 +946,11 @@ def test_multi_iter_frame(self, three_group):
         df["k1"] = np.array(["b", "b", "b", "a", "a", "a"])
         df["k2"] = np.array(["1", "1", "1", "2", "2", "2"])
         grouped = df.groupby(["k1", "k2"])
-        groups = {key: gp for key, gp in grouped}
+        # calling `dict` on a DataFrameGroupBy leads to a TypeError,
+        # we need to use a dictionary comprehension here
+        groups = {
+            key: gp for key, gp in grouped
+        }  # pylint: disable=unnecessary-comprehension
```
Comment (Moisan) [inline pandas/tests/groupby/test_grouping.py:953]: Related question: is it intentional that `dict(df.groupby(["k1", "k2"]))` leads to a `TypeError` while it's possible to iterate on it as if it was a dictionary?

Comment (rhshadrach) [reply pandas/tests/groupby/test_grouping.py:953]: Not exactly intentional, but I don't think there is a good way to make it work. The dict constructor either takes a mapping or an iterable. A groupby object has all the data for a mapping, but doesn't support accessing like a map. We can however use the fact that dict takes an iterable - you should be able to use `dict(grouped.items())`.

Comment (Moisan) [reply pandas/tests/groupby/test_grouping.py:953]: `dict(grouped.items())` leads to 
```
AttributeError: 'DataFrameGroupBy' object has no attribute 'items'
```

Comment (rhshadrach) [reply pandas/tests/groupby/test_grouping.py]: Ah, indeed! A couple of issues with my previous comment.

 - GroupBy implements `__getitem__`, so as far as Python is concerned it is a mapping; however our implementation of this method is not in the sense of a mapping (it is for column selection)
 - GroupBy implements `keys` as an attribute; Python's `dict` construction is expecting `keys` to be a method. So when Python's dict constructor does `grouped.keys()`, it fails.
 - `dict(iter(grouped))` works.

**Pull request merged**